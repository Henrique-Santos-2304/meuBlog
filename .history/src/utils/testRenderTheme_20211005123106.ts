import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(<ThemeProvider>{children}</ThemeProvider> );
};
