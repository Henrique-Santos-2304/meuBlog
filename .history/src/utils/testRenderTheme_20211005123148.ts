import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children: React.ReactElement): RenderResult => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider> );
};
