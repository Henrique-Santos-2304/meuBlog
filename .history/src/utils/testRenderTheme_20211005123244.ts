import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider> );
};
