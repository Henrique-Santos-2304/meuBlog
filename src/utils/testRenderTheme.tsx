import { ThemeProvider } from "styled-components";
import { ChakraProvider } from "@chakra-ui/react";

import { render, RenderResult } from "@testing-library/react";
import theme from "styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(
    <ThemeProvider theme={theme}>
      <ChakraProvider> {children}</ChakraProvider>
    </ThemeProvider>
  );
};
