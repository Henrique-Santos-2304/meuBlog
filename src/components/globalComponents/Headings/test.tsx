import Heading from ".";
import { screen } from "@testing-library/react";
import { renderTheme } from "utils/testRenderTheme";

const props = {
  msg: "teste",
  colorText: "red",
  paddingText: "2rem",
  fontSizeText: "2.4rem",
  marginText: "1rem",
};
describe("<Heading />", () => {
  it("should render ", () => {
    const { container } = renderTheme(<Heading {...props} />);
    expect(container).toMatchSnapshot();
  });
  it("should have a title specified in props ", () => {
    renderTheme(<Heading {...props} />);
    expect(
      screen.getByRole("heading", {
        name: /teste/i,
      })
    ).toBeInTheDocument();
  });
  it("should have a style default  ", () => {
    renderTheme(<Heading msg="teste" />);
    const heading = screen.getByRole("heading", {
      name: /teste/i,
    });
    expect(heading).toHaveStyle({ color: "#f8f8f8" });
    expect(heading).toHaveStyle({ padding: 0 });
    expect(heading).toHaveStyle({ "font-size": "1.4rem" });
    expect(heading).toHaveStyle({ margin: 0 });
  });
  it("should have a style specified in props ", () => {
    renderTheme(<Heading {...props} />);
    const heading = screen.getByRole("heading", {
      name: /teste/i,
    });
    expect(heading).toHaveStyle({ color: "red" });
    expect(heading).toHaveStyle({ padding: "2rem" });
    expect(heading).toHaveStyle({ "font-size": "2.4rem" });
    expect(heading).toHaveStyle({ margin: "1rem" });
  });
});
