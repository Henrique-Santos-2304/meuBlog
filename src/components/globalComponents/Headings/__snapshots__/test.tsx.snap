// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should render  1`] = `
.c0 {
  position: relative;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  color: red;
  padding: 2rem;
  margin: 1rem;
  font-size: 2.4rem;
  font-family: "Roboto",monospace;
}

.c0::before {
  content: "";
  position: absolute;
  top: 50%;
  left: -1.5rem;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  width: 0.4rem;
  height: 90%;
  border-radius: 2rem;
  background: #b7f600;
}

<div>
  <h1
    class="c0"
  >
    teste
  </h1>
</div>
`;
